src/components/settings/SettingsSSOTab/hooks.js

```diff
@@ -1,6 +1,106 @@
 /* eslint-disable import/prefer-default-export */
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useContext } from 'react';
+import { logInfo } from '@edx/frontend-platform/logging';
 import LmsApiService from '../../../data/services/LmsApiService';
+import { SSOConfigContext } from './SSOConfigContext';
+import {
+  updateIdpMetadataURLAction, updateIdpEntryTypeAction, updateEntityIDAction,
+  updateIdpDirtyState,
+} from './data/actions';
+import { updateSamlProviderData } from './utils';
+
+const useIdpState = () => {
+  const {
+    ssoState, dispatchSsoState, setProviderConfig, setCurrentError,
+  } = useContext(SSOConfigContext);
+  const {
+    idp: {
+      metadataURL,
+      entityID,
+      entryType,
+      isDirty,
```

Comment:
does `isDirty` mean is not valid in this context?

Reply:
nope, isDirty is only set when something is changed in this part of the state, it's done in the reducer

Reply:
I use it to avoid sending additional save requests unless there has been a change

Reply:
ooooh I see! Thanks!